recommendations/index.md

```diff
@@ -149,3 +149,53 @@ Although JSON API does not specify the format of date and time fields, it is
 recommended that servers align with ISO 8601. [This W3C
 NOTE](http://www.w3.org/TR/NOTE-datetime) provides an overview of the
 recommended formats.
+=======
+
+## Asynchronous Processing <a href="#asynchronous-processing" id="asynchronous-processing" class="headerlink"></a>
+
+Consider a situation when you need to create a resource and the operation takes long time to complete.
+
+The request **SHOULD** return a status `202 Accepted` with a link in Header/Location.
+
+```text
+HTTP/1.1 202 Accepted
+Location: https://example.com/photos/queue/5234
```

Comment:
Ok, I figured it out! The 202 response should contain a `Content-Location` header that points to the URI of the queue/status resource. From the [HTTP spec](https://tools.ietf.org/html/rfc7231#section-3.1.4.2):

> If Content-Location is included in a 2xx (Successful) response... and its field-value refers to a URI that differs from the effective request URI, then the origin server claims that the URI is an identifier for a different resource corresponding to the enclosed representation... [For a POST request, among others], such a Content-Location indicates that this payload is a representation reporting on the requested action's status and that the same report is available (for future access with GET) at the given URI."

Everything above I said about using a resource-level self link still applies, but `Content-Location` should be sent as well.